tests/simulation/sim_test.go

```diff
@@ -0,0 +1,231 @@
+package simulation_test
```

Comment:
I created a new folder tests. 
Planning to move towards creating subfolders under this, in the future
- simulation 
- e2e 
- ...